xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamFFmpegArchive.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (C) 2018 Arthur Liberman
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DVDInputStreamFFmpegArchive.h"
+#include "ServiceBroker.h"
+#include "pvr/PVRManager.h"
+#include "pvr/PVRStreamProperties.h"
+#include "pvr/addons/PVRClient.h"
+#include "utils/log.h"
+
+CDVDInputStreamFFmpegArchive::CDVDInputStreamFFmpegArchive(const CFileItem& fileitem)
+  : CDVDInputStreamFFmpeg(fileitem),
+    m_client(CServiceBroker::GetPVRManager().GetClient(fileitem))
+{ }
+
+int64_t CDVDInputStreamFFmpegArchive::GetLength()
+{
+  int64_t ret = 0;
+  if (m_client && m_client->GetLiveStreamLength(ret) != PVR_ERROR_NO_ERROR)
+  {
+    Times times = {0};
+    if (GetTimes(times) && times.ptsEnd >= times.ptsBegin)
+      ret = static_cast<int64_t>(times.ptsEnd - times.ptsBegin);
+  }
+  return ret;
+}
+
+bool CDVDInputStreamFFmpegArchive::GetTimes(Times &times)
```

Comment:
> Left-align `*` and `&` to the base type they modify.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types